### PR TITLE
MAINT: Data Prepper opensearch source metrics

### LINKED ISSUE #6391: [DOC] Add Data Prepper opensearch source metrics
**What do you want to do?**
 
- [ x ] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Provide the current list of all metrics within opensearch source plugin

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by chenqi0805**
### Description
This PR adds opensearch source metrics in data prepper

### Issues Resolved
Closes #6391 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by asifsmohammed — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]

File: _data-prepper/pipelines/configuration/sources/opensearch.md [hunk 1]
```diff
@@ -200,6 +200,26 @@ Option | Required | Type    | Description
 `sts_role_arn` | No | String  | The AWS Security Token Service (AWS STS) role to assume for requests to Amazon OpenSearch Service and Amazon OpenSearch Serverless. Default is `null`, which will use the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html).
 `serverless` | No | Boolean | Should be set to `true` when processing from an Amazon OpenSearch Serverless collection. Defaults to `false`.
 
+## Metrics
+
+The `opensearch` source includes the following metrics.
```
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/sources/opensearch.md:205]:
```suggestion
The `opensearch` source includes the following metrics:
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

**Review by Naarcha-AWS — APPROVED** [review]

File: _data-prepper/pipelines/configuration/sources/opensearch.md [hunk 2]
```diff
@@ -200,6 +200,26 @@ Option | Required | Type    | Description
...
+## Metrics
+
+The `opensearch` source includes the following metrics:
+
+### Counters
+
+- `documentsProcessed`: Measures the total number of documents processed by the `opensearch` source plugin.
+- `indicesProcessed`: Measures the total number of indexes processed by the `opensearch` source plugin.
+- `processingErrors`: Measures the total number of errors during index processing by the `opensearch` Source plugin.
+- `credentialsChanged`: Measures the number of times the `opensearch` source refreshes basic credentials (i.e. username/password).
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/sources/opensearch.md:212]:
```suggestion
- `credentialsChanged`: Measures the number of times that the `opensearch` source refreshes basic credentials (username/password).
```

File: _data-prepper/pipelines/configuration/sources/opensearch.md [hunk 3]
```diff
@@ -200,6 +200,26 @@ Option | Required | Type    | Description
...
+- `clientRefreshErrors`: Measures the number of errors encountered when generating a new client due to the `opensearch` source refreshing basic credentials.
+
+### Timers
+
+- `indexProcessingTime`: Measures the latency of processing an index in seconds by the `opensearch` source plugin.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/sources/opensearch.md:217]:
```suggestion
- `indexProcessingTime`: Measures the `opensearch` source plugin index processing latency, in seconds.
```

File: _data-prepper/pipelines/configuration/sources/opensearch.md [hunk 4]
```diff
@@ -200,6 +200,26 @@ Option | Required | Type    | Description
...
+
+### Distribution summaries
+
+- `bytesReceived`: Measures the distribution of incoming document size in bytes received by the `opensearch` source plugin.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/sources/opensearch.md:221]:
```suggestion
- `bytesReceived`: Measures the size distribution of incoming documents, in bytes, received by the `opensearch` source plugin.
```

File: _data-prepper/pipelines/configuration/sources/opensearch.md [hunk 5]
```diff
@@ -200,6 +200,26 @@ Option | Required | Type    | Description
...
+- `bytesProcessed`: Measures the distribution of incoming document size in bytes successfully processed by the `opensearch` source plugin.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/sources/opensearch.md:222]:
```suggestion
- `bytesProcessed`: Measures the size distribution of incoming document, in bytes, successfully processed by the `opensearch` source plugin.
```

Comment (natebower) [inline _data-prepper/pipelines/configuration/sources/opensearch.md:222]: Confirm the technical accuracy of my change both here and on the preceding line.

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS @chenqi0805 Please see my comments and changes and let me know if you have any questions. Thanks!

**Pull request merged**